src/unix/lwt_engine.ml

```diff
@@ -197,6 +197,57 @@ end
 
 class libev_deprecated = libev ()
 
+external unix_fd_to_fd : Unix.file_descr -> int = "luv_fd_of_unix_fd"
+
+class libuv () = object
+  inherit abstract
+
+  val loop = Luv.Loop.default ()
+
+  method loop = loop
+
+  method private cleanup = Luv.Loop.stop loop
+
+  method iter block =
+    try
+      match (block) with
+      | true -> Luv.Loop.run ~loop ~mode:`ONCE () |> ignore
+      | false -> Luv.Loop.run ~loop ~mode:`NOWAIT () |> ignore
+    with exn ->
+      Luv.Loop.stop loop;
+      raise exn
+
+  method private register_readable fd f =
+    let p = Luv.Poll.init ~loop (unix_fd_to_fd fd) in
+    match p with
+    | Ok poll ->
+        let () = Luv.Poll.start poll [`READABLE] (fun _ -> f ()) in
+        lazy(Luv.Poll.stop poll |> ignore)
+    | Error e -> failwith (Printf.sprintf "Could not register fd for read polling, this is probably a error in Lwt, please open a issue on the repo. \nError message: %s" (Luv.Error.strerror e))
```

Comment:
I suggest replacing `strerror` by [`err_name`](https://aantron.github.io/luv/luv/Luv/Error/index.html#val-err_name) (so it prints `EPERM`, etc.), because, in practice, when we are debugging this, we will be looking either at man pages that list e.g. `EPERM`, or in the libuv source code, which also uses these constants. The full error descriptions aren't that useful, since we are not giving "actionable" feedback to "end" users, and meanwhile we end up having to mentally translate them back into error codes.